src/Kmd.Logic.Gateway.Automation/ValidateProduct.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Kmd.Logic.Gateway.Automation.Gateway;
+using YamlDotNet.Serialization;
+
+namespace Kmd.Logic.Gateway.Automation
+{
+    public class ValidateProduct
+    {
+        private IList<PublishResult> publishResults;
+
+        private readonly MediaConfiguration _mediaConfiguration = new MediaConfiguration()
+        {
+            MaxLogoSizeBytes = 1000000,
+            MaxMarkDownDocumentSizeBytes = 1000000,
+            MaxOpenApiSpecSizeBytes = 20971520,
+            AllowedLogoMimeTypes = new[] { ".png", ".jpeg" },
+            AllowedMarkdownDocumentExtension = ".md",
+        };
+
+        public IList<PublishResult> IsProductAndFolderValid(string folderPath)
+        {
+            this.publishResults = new List<PublishResult>();
+
+            if (!Directory.Exists(folderPath))
+            {
+                this.publishResults.Add(new PublishResult { IsError = true, ResultCode = ResultCode.InvalidInput, Message = "Specified folder doesn’t exist" });
+                return this.publishResults;
+            }
+
+            if (!File.Exists(Path.Combine(folderPath, @"publish.yml")))
```

Comment:
When we have validation of API, is this going to be validated there as well?

Reply:
I think, you should prepare Products validation but only for Products. Such root folder validation should be left for some other method.

Reply:
Yes I think.

Reply:
Agreed, added in Publish  class , so that it could be validated before we validate any entity.